cody-shayne/scripts/articleView.js

```diff
@@ -0,0 +1,114 @@
+
+      if ($('#author-filter option[value="' + authorName + '"]').length === 0) {
+        $('#author-filter').append(optionTag);
+      }
+
+      // REVIEW: Similar to the above, but...
+      //       Avoid duplicates! We don't want to append the category name if the select
+      //       already has this category as an option!
+      category = $(this).attr('data-category');
+      optionTag = '<option value="' + category + '">' + category + '</option>';
+      if ($('#category-filter option[value="' + category + '"]').length === 0) {
+        $('#category-filter').append(optionTag);
+      }
+    }
+  });
+};
+
+articleView.handleAuthorFilter = function() {
+  $('#author-filter').on('change', function() {
+    // REVIEW: Inside this function, "this" is the element that triggered the event handler function we're
+    //         defining. "$(this)" is using jQuery to select that element, so we can chain jQuery methods
+    //         onto it.
+    if ($(this).val()) {
+      // DONE: If the select box was changed to an option that has a value, we need to hide all the articles,
+      //       and then show just the ones that match for the author that was selected.
+      //       Use an "attribute selector" to find those articles, and fade them in for the reader.
+      $('article').css('display','none');
+      $('article[data-author="' + this.value + '"]').show();
+    } else {
+      $('article').show();
+      $('.template').hide();
+      // DONE: If the select box was changed to an option that is blank, we should
+      //       show all the articles, except the one article we are using as a template.
+
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = function() {
+  // DONE: Just like we do for #author-filter above, we should handle change events on the #category-filter element.
+  //       When an option with a value is selected, hide all the articles, then reveal the matches.
+  //       When the blank (default) option is selected, show all the articles, except for the template.
+  //       Be sure to reset the #author-filter while you are at it!
+  $('#category-filter').on('change', function(){
+    if ($(this).val()) {
+      $('author-filter').val('');
+      $('article').css('display', 'none');
+      $('article[data-category="' + this.value + '"]').show();
+    } else {
+      $('article').show();
+      $('.template').hide();
+    }
+  })
+};
+
+articleView.handleMainNav = function() {
+  // DONE: Add an event handler to .main-nav elements that will power the Tabs feature.
+  //       Clicking any .tab element should hide all the .tab-content sections, and then reveal the
+  //       single .tab-content section that is associated with the clicked .tab element.
+  //       So: You need to dynamically build a selector string with the correct ID, based on the
+  //       data available to you on the .tab element that was clicked.
+  $('.icon-home').click(function(){
```

Comment:
although this works there should only be one event handler for both .tab elements, instead of having a separate one for each.